### PR TITLE
correctly report timeupdate events for youtube

### PR DESCRIPTION
**Pull request by a-b-r-o-w-n**

### CODE REVIEW

File: src/js/me-shim.js [hunk 1]
```diff
@@ -835,11 +849,17 @@ mejs.YouTubeApi = {
...
-						mejs.YouTubeApi.createEvent(player, pluginMediaElement, 'timeupdate');
-					}, 250);
+
+					var originalSeekTo = settings.pluginMediaElement.pluginApi.seekTo.bind(player);
+
+					settings.pluginMediaElement.pluginApi.seekTo = function(seconds, allowSeekAhead) {
+						originalSeekTo(seconds, allowSeekAhead);
+						// Allow player to update before triggering timeupdate event
+						setTimeout(function() {
+							mejs.YouTubeApi.createEvent(player, pluginMediaElement, 'seeked');
```
Comment (mziwisky) [inline src/js/me-shim.js:868]: the originalSeekTo doesn't emit a `'seeked'` event?

File: src/js/me-shim.js [hunk 2]
```diff
@@ -835,11 +849,17 @@ mejs.YouTubeApi = {
...
+							mejs.YouTubeApi.createEvent(player, pluginMediaElement, 'timeupdate');
+						}, 100);
```
Comment (mziwisky) [inline src/js/me-shim.js:870]: i know we spoke offline about this magic number 100.  i wish we had a more robust solution, but i don't have a suggestion.  one other concern i didn't think of before -- could this cause a confusing sequence of events?  if there's an interval spewing `timeupdate` events every 250ms, it seems possible we'd actually do the `originalSeekTo`, the player would seek, the interval would fire a `timeupdate` for the new time, _then_ this timeout would fire the `seeked`.  so it looks like the time jumped before the seek.  confusing?  a bit.  problematic (e.g. for our analytics)?  i don't know.

**Review by mziwisky — COMMENTED** [review]
couple of questions

File: src/js/me-shim.js (hunk 1 above)
Comment (a-b-r-o-w-n) [reply src/js/me-shim.js:868]: No. The youtube player doesn't emit any events AFAIK. Emitting `seeked` is more for parity here, incase we wanted to hook into that event specifically at some point in the future (thinking insights).

Comment (a-b-r-o-w-n) [reply src/js/me-shim.js:868]: Here are the events the youtube api exposes: https://developers.google.com/youtube/iframe_api_reference#Events

File: src/js/me-shim.js (hunk 2 above)
Comment (a-b-r-o-w-n) [reply src/js/me-shim.js:870]: I don't really have a good response to this. 🤷‍♂️

Comment (mziwisky) [reply src/js/me-shim.js:870]: `A` for effort.

**Review by mziwisky — DISMISSED** [review]

**Comment by a-b-r-o-w-n** [comment]
I found an issue where the interval may never be stopped if the user navigates away without pausing / stopping the video. I'll have a new PS incoming.

**Comment by mziwisky** [comment]
update looks good.  +2

**Comment by kyler-instructure** [comment]
I'm seeing an issue in which a user navigates back to the my collection page while the youtube video is playing and upon returning and playing the video again, the video plays but the timeline and comment timestamp do not update until I either play/pause again or seek.

**Comment by a-b-r-o-w-n** [comment]
@kyler-instructure good catch. I pushed a fix for that.

**Review by mziwisky — APPROVED** [review]
it's even better than it had previously been

**Comment by kyler-instructure** [comment]
QA+1 Looks good now

**Pull request merged**